### ISSUE TITLE
Error when saving sound file

### ISSUE THREAD

**Issue by MrSheridan-SAES** [defold/defold#9210]
### Expected behaviour
Sound file should save without error
### Actual behaviour
Error is generated
### Steps to reproduce
1. Create new sound file and give it a name
2. Select asset in property pane
3. Type new group name in place of "master"
4. CTRL-S to save while cursor is still in the group name box
5. Click x to close file
6. An error occurred:
java.lang.IllegalArgumentException: No implementation of method: :produce-value of protocol: #'internal.graph.types/Evaluation found for class: nil

It seems to require this exact sequence of steps. Clicking out of the group name box first before saving does not produce the error.

<hr/>

<table>
<tr><td>Build time</td><td>2024-06-25T16:26:09.046426</td></tr>
<tr><td>Defold channel</td><td>editor-alpha</td></tr>
<tr><td>Defold editor sha</td><td>3642fa5b700b8d6e74e3f0595e75cd50eff54459</td></tr>
<tr><td>Defold engine sha</td><td>d6882f432beca85d460ec42497888157c356d058</td></tr>
<tr><td>Defold version</td><td>1.9.0</td></tr>
<tr><td>Error</td><td><a href='https://sentry.io/organizations/defold/issues/?query=id%3A"c87f1429b43f48d19d1daa1c6d8d3109"'>c87f1429b43f48d19d1daa1c6d8d3109</a></td></tr>
<tr><td>GPU</td><td>NVIDIA GeForce RTX 2070/PCIe/SSE2</td></tr>
<tr><td>GPU Driver</td><td>4.6.0 NVIDIA 536.99</td></tr>
<tr><td>Java version</td><td>17.0.5+8</td></tr>
<tr><td>OS arch</td><td>amd64</td></tr>
<tr><td>OS name</td><td>Windows 10</td></tr>
<tr><td>OS version</td><td>10.0</td></tr>
<table>

**Comment by britzl** [comment]
```
clojure.core$_cache_protocol_fn.invokeStatic.class java.lang.IllegalArgumentException: No implementation of method: :produce-value of protocol: #'internal.graph.types/Evaluation found for class: nil
    at clojure.core$_cache_protocol_fn.invokeStatic(core_deftype.clj:583)
    at internal.graph.types$fn$G__6272__6286.invoke(types.clj:87)
    at internal.node$node_value.invokeStatic(node.clj:454)
    at internal.system$node_value.invokeStatic(system.clj:354)
    at dynamo.graph$do_node_value.invokeStatic(graph.clj:840)
    at dynamo.graph$node_value.invokeStatic(graph.clj:843)
    at editor.cljfx_form_view$create_renderer$fn.invoke(cljfx_form_view.clj:1518)
    at cljfx.event_handler$wrap_co_effects$fn$fn.invoke(event_handler.clj:11)
    at clojure.core.protocols$iter_reduce.invokeStatic(protocols.clj:49)
    at clojure.core.protocols$fn.invokeStatic(protocols.clj:75)
    at clojure.core.protocols$fn.invoke(protocols.clj:75)
    at clojure.core.protocols$fn$G__8068__8086.invoke(protocols.clj:13)
    at clojure.core$reduce.invokeStatic(core.clj:6828)
    at cljfx.event_handler$wrap_co_effects$fn.invoke(event_handler.clj:10)
    at cljfx.event_handler$wrap_effects$dispatch_sync_BANG___25356.invoke(event_handler.clj:27)
    at cljfx.event_handler$wrap_effects$dispatch_sync_BANG___25356.invoke(event_handler.clj:25)
    at editor.cljfx_form_view$wrap_force_refresh$fn.invoke(cljfx_form_view.clj:1509)
    at cljfx.lifecycle$make_handler_fn$fn.invoke(lifecycle.clj:131)
    at cljfx.coerce$change_listener$reify.changed(coerce.clj:490)
    at com.sun.javafx.binding.ExpressionHelper$SingleChange.fireValueChangedEvent(ExpressionHelper.java:192)
    at com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:91)
    at javafx.beans.property.ObjectPropertyBase.fireValueChangedEvent(ObjectPropertyBase.java:106)
    at javafx.beans.property.ObjectPropertyBase.markInvalid(ObjectPropertyBase.java:113)
    at javafx.beans.property.ObjectPropertyBase.set(ObjectPropertyBase.java:147)
    at javafx.scene.control.TextFormatter.setValue(TextFormatter.java:179)
    at javafx.scene.control.TextFormatter.updateValue(TextFormatter.java:206)
    at javafx.scene.control.TextInputControl.commitValue(TextInputControl.java:1330)
    at javafx.scene.control.TextInputControl.lambda$new$3(TextInputControl.java:198)
    at com.sun.javafx.binding.ExpressionHelper$Generic.fireValueChangedEvent(ExpressionHelper.java:372)
    at com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:91)
    at javafx.beans.property.ReadOnlyBooleanPropertyBase.fireValueChangedEvent(ReadOnlyBooleanPropertyBase.java:78)
    at javafx.scene.Node$FocusPropertyBase.notifyListeners(Node.java:8175)
    at javafx.scene.Node$17.notifyListeners(Node.java:8263)
    at javafx.scene.Node.notifyFocusListeners(Node.java:8196)
    at javafx.scene.Scene$FocusOwnerProperty.invalidated(Scene.java:2256)
    at javafx.beans.property.ObjectPropertyBase.markInvalid(ObjectPropertyBase.java:112)
    at javafx.beans.property.ObjectPropertyBase.set(ObjectPropertyBase.java:147)
    at javafx.scene.Scene.setFocusOwner(Scene.java:2307)
    at javafx.scene.Scene.requestFocus(Scene.java:2202)
    at javafx.scene.Scene$ScenePulseListener.focusCleanup(Scene.java:2564)
    at javafx.scene.Scene$ScenePulseListener.pulse(Scene.java:2583)
    at com.sun.javafx.tk.Toolkit.lambda$runPulse$2(Toolkit.java:401)
    at java.security.AccessController.doPrivileged(AccessController.java:399)
    at com.sun.javafx.tk.Toolkit.runPulse(Toolkit.java:400)
    at com.sun.javafx.tk.Toolkit.firePulse(Toolkit.java:430)
    at com.sun.javafx.tk.quantum.QuantumToolkit.pulse(QuantumToolkit.java:598)
    at com.sun.javafx.tk.quantum.QuantumToolkit.pulse(QuantumToolkit.java:578)
    at com.sun.javafx.tk.quantum.QuantumToolkit.pulseFromQueue(QuantumToolkit.java:571)
    at com.sun.javafx.tk.quantum.QuantumToolkit.lambda$runToolkit$11(QuantumToolkit.java:352)
    at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
    at com.sun.glass.ui.win.WinApplication._runLoop(WinApplication.java)
    at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:184)
    at java.lang.Thread.run(Thread.java:833)
```

**Comment by vlaaad** [comment]
Couldn't reproduce it on a mac, maybe Windows-specific?

**Comment by AGulev** [comment]
This issue looks outdated: we can't reproduce it and also it's hard to follow old stack trace.
I'm closing it for now. If you are able to reproduce this issue using Defold 1.12.2+ please feel free to reopen it. 

## Issue summary

The issue reports an editor exception when a `.sound` file is saved while the cursor is still active in the `group` text field, followed by closing the file. The original report says the failure only happens with that exact save-while-editing sequence, and the only follow-up comments were a stack trace plus a note that it could not be reproduced on macOS.

## Status

Possibly outdated/fixed

## Evidence

- GitHub still shows the issue as `OPEN`, labeled `bug` and `editor`, with no closing comment or linked fix.
- The original stack trace in the issue points into `editor.cljfx_form_view/create_renderer`, i.e. the older form/property editing path, not the current dedicated properties pane implementation.
- The current editor wires the right-side properties pane through `editor/src/clj/editor/boot_open_project.clj:193` and `editor/src/clj/editor/boot_open_project.clj:375`, which create and connect `editor.properties-view` instead of routing property edits through `editor.cljfx_form_view`.
- `editor/src/clj/editor/properties_view.clj:804` shows the active properties pane node is now `PropertiesView`, which is a different code path from the one named in the issue stack trace.
- After the issue was filed, commit `30b5ee3e07358d50ef32527a9ed9f65d6b6209b8` updated `editor/src/clj/editor/properties_view.clj` and `editor/src/clj/editor/fxui.clj` with the note `Fix override coalesce and stale handler context bugs in properties-view`, which is consistent with the kind of focused-field/save/close timing problem described in #9210.
- `editor/src/clj/editor/sound.clj:233` still allows `.sound` files to be opened in a form view tab, so I cannot prove every old code path is gone, but the exact "property pane" workflow from the report now appears to use the newer properties-view implementation.

## Why I think it is already fixed

The report is specifically about editing a sound property in the property pane and then saving/closing while the text field still has focus. In the current codebase, that pane is no longer driven by the old `cljfx_form_view` path named in the exception stack. It now goes through `editor.properties-view`, and the post-issue history includes an explicit fix for stale handler context bugs in that newer properties UI layer. I did not find a dedicated regression test for the exact sound-field reproduction sequence, so I would not mark it definitely fixed, but the original failure mode looks likely to be outdated.
